### PR TITLE
encoding functions returns non-empty-string

### PR DESCRIPTION
**Pull request by staabm**
analog https://github.com/vimeo/psalm/pull/8431

fixes https://phpstan.org/r/3a8a8430-1d4c-4bb9-99a8-f3838932f1e6

### CODE REVIEW

**Comment by staabm** [comment]
the error in composer is valid and it was there before this PR:

```
 ------ --------------------------------------------------------------------- 
  Line   src/Composer/Util/RemoteFilesystem.php                               
 ------ --------------------------------------------------------------------- 
  327    Parameter #1 $string of function base64_encode expects T of string,  
         string|false given.                                                  
 ------ --------------------------------------------------------------------- 
 ```

since the error message changed because generics are involved now, the already existing ignore-line no longer mutes it

https://github.com/ondrejmirtes/composer/blob/acef69614d5c54def3418d3964749b49923200b2/phpstan/baseline-8.1.neon#L278-L281

File: stubs/core.stub [hunk 1]
```diff
@@ -47,3 +47,31 @@ function highlight_string($var, bool $return = false) {}
  * @return ($return is true ? string : true)
  */
 function print_r($var, bool $return = false) {}
+
+/**
+ * @template T as string
+ * @param T $string
+ * @return (T is non-empty-string ? non-empty-string : string)
```
Comment (ondrejmirtes) [inline stubs/core.stub]: You don't need generics for this, `($string is non-empty-string ? non-empty-string : string)` should work just as well. Please edit all of these fuctions.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: stubs/core.stub (hunk 1 above)
Comment (staabm) [reply stubs/core.stub]: good catch, fixed.

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**